src/engine/Server.cpp

```diff
@@ -344,27 +344,26 @@ Awaitable<void> Server::process(
   checkParameterNotPresent("default-graph-uri");
   checkParameterNotPresent("named-graph-uri");
 
-  auto checkParameter = [&parameters](std::string_view key,
-                                      std::optional<std::string> value,
-                                      bool accessAllowed = true) {
-    std::optional<string> param =
-        Server::checkParameter(parameters, key, std::move(value));
-    // Now that we have the value, check if there is a problem with the access.
-    // If yes, we abort the query processing at this point.
-    if (param && !accessAllowed) {
-      throw std::runtime_error(absl::StrCat("Access to \"", key, "=",
-                                            param.value(), "\" denied",
-                                            " (requires a valid access token)",
-                                            ", processing of request aborted"));
-    }
-    return param;
-  };
+  // We always want to call `Server::checkParameter` with the same first
+  // parameter.
+  auto checkParameter =
+      std::bind(&Server::checkParameter, std::cref(parameters),
+                std::placeholders::_1, std::placeholders::_2);
```

Comment:
```suggestion
      std::bind_front(&Server::checkParameter, std::cref(parameters));
```
(std::bind is deprecated).